src/go/rpk/pkg/tuners/irq/cpu_masks.go

```diff
@@ -83,6 +85,21 @@ func (masks *cpuMasks) CPUMaskForComputations(
 	} else if mode == Mq {
 		// all available cores
 		computationsMask = cpuMask
+	} else if mode == Dedicated {
+		numOfCores, err := masks.GetNumberOfPUs(cpuMask)
+		if err != nil {
+			return "", err
+		}
+		rpCores := numOfCores - uint(math.Ceil(float64(numOfCores)/float64(t.CoresPerDedicatedInterruptCore)))
```

Comment:
So what happens if rpCores == numOfCores? We still go through the "dedicated" flow but with 0 dedicated cores? Should we instead switch to "mq" mode or something? To reduce the state space.

Reply:
> I think we should just error out as per above. Note the only place where we can switch mode is in GetDefaultMode. We can't just "switch mode" in any of the other functions as things will become inconsistent then.

I see, I was thinking this was still very near the start and maybe we could do that.